### ISSUE TITLE
segfault in NECI tests

### ISSUE THREAD

**Issue by boegel** [easybuilders/easybuild-easyconfigs#17164]
The test step of `NECI-20220711-foss-2022a.eb` fails with segfaults when not enough memory is available:

```
The following tests FAILED:
         27 - test_neci_back_spawn_excit_gen (SEGFAULT)
         28 - test_kneci_back_spawn_excit_gen (SEGFAULT)
         29 - test_dneci_back_spawn_excit_gen (SEGFAULT)
         30 - test_mneci_back_spawn_excit_gen (SEGFAULT)
         31 - test_kdneci_back_spawn_excit_gen (SEGFAULT)
         32 - test_kmneci_back_spawn_excit_gen (SEGFAULT)
```

I saw this when ~44GB or even ~88GB is available.

**Comment by boegel** [comment]
I've tried forcing that only one test at a time is run by using `ctest -j 1` instead of `ctest -j`, but that doesn't help...

Any ideas here @lexming?

Can we do better than somehow producing a helpful error message to suggest doing the installation in an environment with more available memory if the test suite failed?

**Comment by boegel** [comment]
Seeing this again when building NECI with GCC 12.3.0 (`foss/2023a`), via #21616, so I've reported this upstream:
* https://github.com/ghb24/NECI_STABLE/issues/18

**Comment by lexming** [comment]
Fixed in #21616 with patch:
https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/n/NECI/NECI-20230620_segfault.patch
